Frameworks/Foundation/NSURLProtocol.mm

```diff
@@ -113,28 +117,27 @@ + (BOOL)canInitWithRequest:(NSURLRequest*)request {
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 + (id)propertyForKey:(NSString*)key inRequest:(NSURLRequest*)request {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [request _propertyForKey:key];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 + (void)setProperty:(id)value forKey:(NSString*)key inRequest:(NSMutableURLRequest*)request {
-    UNIMPLEMENTED();
+    [request _setProperty:value forKey:key];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 + (void)removePropertyForKey:(NSString*)key inRequest:(NSMutableURLRequest*)request {
-    UNIMPLEMENTED();
+    [request _removePropertyForKey:key];
```

Comment:
It might be worth checking whether this works on immutable requests... even though this is specified for `NSMutableURLRequest*`, we'll want to figure out what the error case looks like